UnityAds/src/main/java/com/mopub/mobileads/UnityInterstitial.java

```diff
@@ -45,28 +70,27 @@ protected void load(@NonNull final Context context, @NonNull final AdData adData
         mContext = context;
 
         setAutomaticImpressionAndClickTracking(false);
-        UnityAds.load(mPlacementId);
-
-        mUnityAdsAdapterConfiguration.setCachedInitializationParameters(context, extras);
 
-        UnityRouter.getInterstitialRouter().addListener(mPlacementId, this);
         UnityRouter.getInterstitialRouter().setCurrentPlacementId(mPlacementId);
-        initializeUnityAdsSdk(extras);
-    }
 
-    private void initializeUnityAdsSdk(Map<String, String> serverExtras) {
-        if (!UnityAds.isInitialized()) {
-            if (!(mContext instanceof Activity)) {
-                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Context is null or is not an instanceof Activity.");
-                return;
+        UnityRouter.initUnityAds(extras, context, new IUnityAdsInitializationListener() {
+            @Override
+            public void onInitializationComplete() {
+                UnityAds.load(mPlacementId, mUnityLoadListener);
             }
-            UnityRouter.initUnityAds(serverExtras, (Activity) mContext);
-        }
+
+            @Override
+            public void onInitializationFailed(UnityAds.UnityAdsInitializationError unityAdsInitializationError, String errorMessage) {
+                MoPubLog.log(CUSTOM, ADAPTER_NAME, errorMessage);
+            }
+        });
+        mUnityAdsAdapterConfiguration.setCachedInitializationParameters(context, extras);
     }
 
     @Override
     protected void show() {
         MoPubLog.log(SHOW_ATTEMPTED, ADAPTER_NAME);
+        UnityRouter.getInterstitialRouter().addListener(mPlacementId, this);
```

Comment:
move this line inside` if(UnityAds.isReady)` otherwise, it wouldn't be able to get cleaned.

Reply:
Also, we probably could add/remove listener to UnityAds directly, rather than through UnityRouter.